sklearn/utils/estimator_checks.py

```diff
@@ -1520,6 +1521,20 @@ def check_sample_weights_list(name, estimator_orig):
     estimator.fit(X, y, sample_weight=sample_weight)
 
 
+def check_all_zero_sample_weights_error(name, estimator_orig):
+    """Check that estimator raises error when all sample weights are 0."""
+    estimator = clone(estimator_orig)
+
+    X, y = make_classification(random_state=42)
+    X = _enforce_estimator_tags_X(estimator, X)
+    y = _enforce_estimator_tags_y(estimator, y)
+
+    sample_weight = np.zeros(_num_samples(X))
+
+    with raises(ValueError, match=r"(.*weight.*zero.*)|(.*zero.*weight.*)"):
```

Comment:
Let's add a comment above listing the 3 estimators and the alternative message each of these give.